File: engine/engine-model/src/main/scala/tools/packet/buff/ShowMonsterRiding.scala
package tools.packet.buff

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowMonsterRiding(private var _characterId: Int, private var _mountId: Int, private var _skillId: Int) extends PacketInput {
  def characterId: Int = _characterId

  def mountId: Int = _mountId

  def skillId: Int = _skillId

  override def opcode(): SendOpcode = SendOpcode.GIVE_FOREIGN_BUFF
}

File: engine/engine-model/src/main/java/tools/packet/buff/ShowMonsterRiding.java
package tools.packet.buff;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowMonsterRiding(Integer characterId, Integer mountId, Integer skillId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GIVE_FOREIGN_BUFF;
   }
}
